src/components/controls/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,109 @@
+'use client';
+
+import React from 'react';
+
+import {Eye, EyeSlash} from '@gravity-ui/icons';
+
+import {useControlledState} from '../../../hooks';
+import {ActionTooltip} from '../../ActionTooltip';
+import {Button} from '../../Button';
+import {ClipboardButton} from '../../ClipboardButton';
+import {Icon} from '../../Icon';
+import {block} from '../../utils/cn';
+import {TextInput} from '../TextInput';
+import type {TextInputProps} from '../TextInput';
+
+import {i18n} from './i18n';
+import {getActionButtonSizeAndIconSize} from './utils';
+
+import './PasswordInput.scss';
+
+const b = block('password-input');
+
+export type PasswordInputProps = Omit<TextInputProps, 'type'> & {
+    /** Show copy button */
+    showCopyButton?: boolean;
+    /** Show reveal button */
+    showRevealButton?: boolean;
+    /** Disable the tooltip for the copy button. The tooltip will not be displayed */
+    hasCopyTooltip?: boolean;
+    /** Disable the tooltip for the reveal button. The tooltip will not be displayed */
+    hasRevealTooltip?: boolean;
+};
+
+export const PasswordInput = (props: PasswordInputProps) => {
+    const {
+        autoComplete,
+        showCopyButton,
+        rightContent,
+        endContent,
+        showRevealButton,
+        size = 'm',
+        hasCopyTooltip = true,
+        hasRevealTooltip = true,
+        controlProps,
+    } = props;
+
+    const [inputValue, setInputValue] = useControlledState(
+        props.value,
+        props.defaultValue ?? '',
+        props.onUpdate,
+    );
+
+    const [hideValue, setHideValue] = React.useState(true);
```

Comment:
Let's also allow change visibility via props. @amje

Reply:
give full control over changing the value hiding? Or give the ability to throw the initial state?

Reply:
full control, default value seems redundant.

Reply:
maybe then we should make two options with full control or without it as now?

Reply:
Yes, I meant that.
```
const [revealValue, setRevealValue] = useControlledState(props.revealValue, false, props.onRevealValueUpdate);
```